docs/velocity/developers/api/command.md

```diff
@@ -22,31 +22,81 @@ and dispatch command actions. You can register your own `CommandNode`s by wrappi
 import net.kyori.adventure.text.format.NamedTextColor;
 
-@Plugin(id = "helloworld")
-public class HelloWorldPlugin {
+public final class TestBrigadierCommand {
 
-    public void createBrigadierCommand() {
+    public static BrigadierCommand createBrigadierCommand(final ProxyServer proxy) {
         LiteralCommandNode<CommandSource> helloNode = LiteralArgumentBuilder
             .<CommandSource>literal("test")
+            // Here you can filter the subjects that can execute the command.
+            // This is the ideal place to do "hasPermission" checks
+            .requires(source -> source.hasPermission("test.permission"))
+            // Here you can add the logic that will be used in
+            // the execution of the "/test" command without any argument
             .executes(context -> {
+                // Here you get the subject that executed the command
+                CommandSource source = context.getSource();
+
                 Component message = Component.text("Hello World", NamedTextColor.AQUA);
-                context.getSource().sendMessage(message);
-                return 1; // indicates success
+                source.sendMessage(message);
+
+                // Returning Command.SINGLE_SUCCESS means that the execution was successful
+                // Returning BrigadierCommand.FORWARD will send the command to the server
+                return Command.SINGLE_SUCCESS;
             })
+            // Using the "then" method you can add subarguments to the command.
+            // For example, this subcommand will be executed when using the command "/test <some argument>"
+            // A RequiredArgumentBuilder is a type of argument in which you can enter some undefined data
+            // of some kind. For example, this example uses a StringArgumentType.word() that requires
+            // a single word to be entered, but you can also use different ArgumentTypes provided by Brigadier
+            // that return data of type Boolean, Integer, Float, other String types, etc
+            .then(RequiredArgumentBuilder.<CommandSource, String>argument("argument", StringArgumentType.word())
+                // Here you can define the hints to be provided in case the ArgumentType does not provide them.
+                // In this example the names of all connected players are provided
+                .suggests((ctx, builder) ->
+                    // Here we provide the names of the players along with a tooltip,
+                    // which can be used as an explanation of a specific argument or as a simple decoration
+                    proxy.getAllPlayers().forEach(player -> builder.suggest(
+                        player.getUsername(),
+                        // A VelocityBrigadierMessage receives as argument a component,
+                        // in this case the name of the player is provided with a rainbow
+                        // gradient thanks to MiniMessage (Library available since Velocity 3.1.2+)
```

Comment:
I don't like my wording too much here either, but both "receives as argument a component" and the comma make this a bit hard to read. hopefully, this is better.
```suggestion
                        // A VelocityBrigadierMessage takes a component.
                        // In this case, the player's name is provided with a rainbow
                        // gradient created by MiniMessage (Library available since Velocity 3.1.2+)
```